test/built-ins/Promise/prototype/finally/invokes-then-with-function.js

```diff
@@ -43,11 +44,16 @@ assert.sameValue(
   'Invokes `then` method with a function as the first argument'
 );
 assert.notSameValue(firstArg, originalFinallyHandler, 'Invokes `then` method with a different fulfillment handler');
+assert.sameValue(firstArg.length, 1, 'fulfillment handler has a length of 1');
+assert.sameValue(firstArg.name, anonName, 'fulfillment handler is anonymous');
+
 assert.sameValue(
   typeof secondArg,
   'function',
   'Invokes `then` method with a function as the second argument'
 );
-assert.notSameValue(secondArg, originalFinallyHandler, 'Invokes `then` method with a different fulfillment handler');
+assert.notSameValue(secondArg, originalFinallyHandler, 'Invokes `then` method with a different rejection handler');
```

Comment:
nit: a different handler, used for rejection.

Reply:
while you're there, it might sound a bit silly but you could add an assertion confirming `firstArg !== secondArg`. 

It's also interesting to check that both are functions directly built from `Function.prototype` using `Object.getPrototypeOf` and are not from any subclass, which would give a false positive for typeof.

Reply:
hm, subclasses can't give a false positive for `typeof`; typeof uses the internal [[Call]] slot

Reply:
```js
class F extends Function {}
Object.getPrototypeOf(F) === Function.prototype; // false
typeof F; // "function"
```

Reply:
that's not a false positive for `typeof` tho; what `typeof` is checking is that you can do `F()`, which you can.

The prototype check, `assert.sameValue(Object.getPrototypeOf(F), Function.prototype)` - something that doesn't seem to be commonly done - checks something different.